server/sonar-server/src/main/java/org/sonar/server/component/db/ComponentDao.java

```diff
@@ -182,18 +181,36 @@ protected void doDeleteByKey(DbSession session, String key) {
   public List<ComponentDto> selectProvisionedProjects(DbSession session, SearchOptions searchOptions, @Nullable String query) {
     Map<String, String> parameters = new HashMap<>();
     parameters.put("qualifier", Qualifiers.PROJECT);
-    if (query != null) {
-      parameters.put("query", "%" + query + "%");
-    }
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
     return mapper(session).selectProvisionedProjects(parameters, new RowBounds(searchOptions.getOffset(), searchOptions.getLimit()));
   }
 
   public int countProvisionedProjects(DbSession session, @Nullable String query) {
     Map<String, String> parameters = new HashMap<>();
     parameters.put("qualifier", Qualifiers.PROJECT);
-    if (query != null) {
-      parameters.put("query", "%" + query + "%");
-    }
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
     return mapper(session).countProvisionedProjects(parameters);
   }
+
+  public List<ComponentDto> selectGhostProjects(DbSession session, @Nullable String query, SearchOptions options) {
+    Map<String, String> parameters = new HashMap<>();
```

Comment:
in this case, shouldn't we use http://guava-libraries.googlecode.com/svn-history/r14/trunk/javadoc/com/google/common/collect/Maps.html#newHashMapWithExpectedSize%28int%29 ?